plugins/mtd/fu-plugin-mtd.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2021 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+void
+fu_plugin_init(FuPlugin *plugin)
+{
+	fu_plugin_add_udev_subsystem(plugin, "mtd");
+	fu_plugin_set_build_hash(plugin, FU_BUILD_HASH);
+	fu_plugin_add_device_gtype(plugin, FU_TYPE_MTD_DEVICE);
+}
+
+gboolean
+fu_plugin_startup(FuPlugin *plugin, GError **error)
+{
+#ifndef HAVE_MTD_USER_H
+	g_set_error_literal(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_NOT_SUPPORTED,
+			    "Not supported as mtd-user.h is unavailable");
+	return FALSE;
+#endif
+	return TRUE;
+}
+
+gboolean
+fu_plugin_device_created(FuPlugin *plugin, FuDevice *dev, GError **error)
+{
+	fu_device_set_summary(dev, "Memory Technology Device");
+	fu_device_add_protocol(dev, "org.infradead.mtd");
+	fu_device_add_flag(dev, FWUPD_DEVICE_FLAG_INTERNAL);
+	fu_device_add_flag(dev, FWUPD_DEVICE_FLAG_NEEDS_REBOOT);
+	fu_device_add_vendor_id(dev, "XXX");
```

Comment:
TODO I guess

Reply:
I've changed this to FU_HWIDS_KEY_MANUFACTURER